guard/src/rules/errors.rs

```diff
@@ -1,173 +1,68 @@
-        }
-
-        ErrorKind::FormatError(fmt) => {
-            format!("Formatting error when writing {}", fmt)
-        }
-
-        ErrorKind::IoError(io) => {
-            format!("I/O error when reading {}", io)
-        }
-
-        ErrorKind::ParseError(err) => {
-            format!("Parser Error when parsing {}", err)
-        }
-
-        ErrorKind::RegexError(err) => {
-            format!("Regex expression parse error for rules file {}", err)
-        }
-
-        ErrorKind::MissingProperty(err) => {
-            format!("Could not evaluate clause for a rule with missing property for incoming context {}", err)
-        }
-
-        ErrorKind::MissingVariable(err) => {
-            format!(
-                "Variable assignment could not be resolved in rule file or incoming context {}",
-                err
-            )
-        }
-
-        ErrorKind::MultipleValues(err) => {
-            format!(
-                "Conflicting rule or variable assignments inside the same scope {}",
-                err
-            )
-        }
-
-        ErrorKind::IncompatibleRetrievalError(err) => {
-            format!(
-                "Types or variable assignments have incompatible types to retrieve {}",
-                err
-            )
-        }
-
-        IncompatibleError(err) => {
-            format!("Types or variable assignments are incompatible {}", err)
-        }
-
-        ErrorKind::NotComparable(err) => {
-            format!(
-                "Comparing incoming context with literals or dynamic results wasn't possible {}",
-                err
-            )
-        }
-
-        ErrorKind::ConversionError(_ignore) => "Could not convert in JSON value object".to_string(),
-
-        ErrorKind::Errors(all) => {
-            let vec = all.iter().map(error_kind_msg).collect::<Vec<String>>();
-            format!("{:?}", &vec)
-        }
-
-        ErrorKind::RetrievalError(err) => {
-            format!(
-                "Could not retrieve data from incoming context. Error = {}",
-                err
-            )
-        }
-
-        ErrorKind::MissingValue(err) => {
-            format!(
-                "There was no variable or value object to resolve. Error = {}",
-                err
-            )
-        }
-
-        ErrorKind::FileNotFoundError(path) => {
-            format!("The path {} does not exist", path)
-        }
-    }
-}
-
-fn error_kind_fmt(kind: &ErrorKind, f: &mut Formatter<'_>) -> std::fmt::Result {
-    writeln!(f, "{}", error_kind_msg(kind))
-}
-
-impl Display for Error {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        error_kind_fmt(&self.0, f)
-    }
-}
-
-#[derive(Debug)]
-pub enum ErrorKind {
-    JsonError(serde_json::Error),
-    YamlError(serde_yaml::Error),
-    FormatError(std::fmt::Error),
-    IoError(std::io::Error),
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Error parsing incoming JSON context")]
```

Comment:
Do these also add the error message from the internal error? The old behavior appended the underlying message right? With ...`{}, err`...

Reply:
yes, they do. Thats what the #[from] macro does